content/anchor-frontend.md

```diff
@@ -0,0 +1,619 @@
+# Invoke Anchor Program from Frontend
+
+# Lesson Objectives
+
+*By the end of this lesson, you will be able to:*
+
+- Explain `IDL`
+- Explain an Anchor `Provider`
+- Explain an Anchor `Program`
+- Use the Anchor `MethodsBuilder` to build transactions
+- Set up a frontend to invoke instructions from a Solana program built using Anchor
+
+# TL;DR
+
+- An `IDL` is a file representing the structure of a program that Anchor generates when the program is built
+- An Anchor `Provider` is an object that combines a `connection` to a cluster and a specified `wallet` to sign transactions
+- An Anchor `Program` is an object that provides a custom API to interact with a specific program by combining a program `IDL` and `Provider`.
+- The Anchor `MethodsBuilder` provides a simplified format for building transactions with instructions from a `Program`.
+- `@project-serum/anchor` is a TypeScript client that includes everything you’ll need to interact with Anchor programs
+
+# Overview
+
+Anchor simplifies the process of interacting with programs from the client by providing an `IDL` that reflects the structure of a program. Using the `IDL` along with the Anchor `MethodsBuilder` provides a simplified format for building transactions.
+
+```tsx
+// sends transaction
+await program.methods
+  .instructionName(instructionDataInputs)
+  .accounts({})
+  .signers([])
+  .rpc()
+```
+
+In this lesson we will go over how to invoke an Anchor program from a frontend. We’ll use `@project-serum/anchor`, a TypeScript client for Anchor programs, which includes everything we’ll need to interact with a program from the client.
+
+## Anchor Setup
+
+Before we can interact with a program using a frontend, we’ll need to create an Anchor `Program` object. The `Program` object provides a custom API to interact with a specific program by combining a program `IDL` and `Provider`.
+
+To create the `Program` object, we’ll need the following:
+
+- `IDL` - file representing the structure of a program
+- `Connection` - the cluster connection
+- `Wallet` - default keypair used to pay for and sign transactions
+- `Provider` - encapsulates the `Connection` to a Solana cluster and a `Wallet`
+- `ProgramId` - the program’s on-chain address
+
+Next, let’s go over each item to better understand how everything ties together.
+
+### IDL (Interface Description Language)
```

Comment:
Can you generate an IDL from a non-anchor program? I think you can but not sure how. I think it would be helpful to talk about that a little here, even if you can't maybe just mention that you can't interact with a non-Anchor program in the same way that you can an Anchor program with the `@project-serum/anchor` package.

Reply:
also, APR (anchor program registry) is still pretty new but I think it'd be worth mentioning here as well. It's a public repository of program IDL's and such. Made by Coral/Armani/Italo 

https://www.apr.dev/

Reply:
I there are 3rd party tools people have created that generate IDLs out of native Solana programs, but I've never used them.

https://github.com/metaplex-foundation/shank
https://github.com/acheroncrypto/native-to-anchor

Reply:
how is APR supposed to be used?